internal/providers/openstack/openstack.go

```diff
@@ -167,13 +178,61 @@ func fetchConfigFromDevice(logger *log.Logger, ctx context.Context, path string)
 }
 
 func fetchConfigFromMetadataService(f *resource.Fetcher) ([]byte, error) {
-	res, err := f.FetchToBuffer(metadataServiceUrl, resource.FetchOptions{})
+	urls := map[string]url.URL{
+		string(resource.IPv4): userdataURLs[resource.IPv4],
+	}
+
+	ifaceName, err := findInterfaceWithIPv6()
```

Comment:
Now since we have a list of strings, rather then 1 lets reach out to all of them at the same time. 

I think we can do this a few ways, though I think it would be cool to do it in parallel using [fan-out](https://ornlu-is.github.io/go_fan_out_pattern/) pattern.

Golang has a few concepts that will help us to achieve that.

[goroutines](https://gobyexample.com/goroutines), [channels](https://gobyexample.com/channels), [waitgroups](https://gobyexample.com/waitgroups)